### DISCUSSION TITLE
Can't get window name or class for gifine (lua script) app to avoid fullscreen blur

### DISCUSSION THREAD

**Discussion by Flashwalker** [yshui/picom#1013] [Q&A]
There is a [gifine](https://github.com/leafo/gifine) app for screen record. It's a lua script. It uses [xrectsel](https://github.com/ropery/xrectsel) script to select a part of the screen. During the selection a fullscreen transparent window appears and it's blured, and no way to **xprop** it, because it grabs the mouse.
Any ideas?

For **gifine** window (not that fullscreen transparent window) class is:
WM_CLASS(STRING) = "lua5.1", "Lua5.1"

i tried to exclude blur for:
```
    ,"class_g = 'lua5.1'"
    ,"class_g = 'Lua5.1'"
    ,"class_i = 'lua5.1'"
    ,"class_i = 'Lua5.1'"
    ,"name *= 'Lua5.1'"
    ,"name *= 'lua5.1'"
    ,"name ~= 'Lua5.1'"
    ,"name ~= 'lua5.1'"
```

nothing helps

https://user-images.githubusercontent.com/2188215/215327858-86518d8e-c5be-4f86-bd83-66f658fca31b.mp4

**Comment by Flashwalker** [comment]
The only workaround i found for now is to whitelist certain apps and exclude the rest:
E.g.:
```
# Blur: whitelist:
blur-background-exclude = [
  "class_g != 'URxvt' &&
   class_g != 'Alacritty' &&
   class_g != 'st-256color' &&
   class_g != 'tabbed' &&
   class_g != 'kitty' &&
   class_g != 'Xfce4-terminal' &&
   class_g != 'Notification' &&
   window_type != 'tooltip' &&
   window_type != 'menu' &&
   window_type != 'popup_menu' &&
   window_type != 'dropdown_menu' &&
   window_type != 'splash' &&
   window_type != 'combo'"
   
   ,"name = 'Whisker Menu'"
   ,"name = 'xfce4-panel-plug'"
];
```

**Comment by Flashwalker** [comment]
or just:
```
wintypes : 
{
  normal = {
    blur-background = false;
  };
};
```

**Comment by absolutelynothelix** [comment]
are you sure that your `gifine` uses `xrectsel`, not `slop`? if it really uses `xrectsel`, i'd recommend you to switch to `slop`. if it uses `slop`, it's window can be excluded with `class_i = 'slop'`.

**Reply by Flashwalker** [reply]
thanks, i'll try
